src/attachments/modifyImage.spec.js

```diff
@@ -291,6 +292,123 @@ describe("image modification", () => {
       }));
   });
 
+  it("will print the progress status when trying to modify the same image again", function () {
+    this.timeout(30 * 1000);
+
+    const tmpDir = tmp.dirSync({unsafeCleanup: true});
+    const outPath = tmpDir.name;
+    const database = new Database(`${outPath}/database.json`);
+    const images = [fixtureFile, fixtureFile2, fixtureFile];
+    const [image1, image2, image3] = images;
+
+    let counter = 0;
+
+    return cleanUpWhenDone(tmpDir)(modifyImages({
+      database,
+      key: "test",
+      outPath,
+      imageWidth: 10,
+      minify: true,
+      progress: ({message, currentStep, steps}) => {
+        counter++;
+
+        expect(steps).to.be(3);
+
+        if (counter === 1) {
+          // "init" message
+          expect(message).not.to.be.empty();
+          expect(currentStep).to.be(0);
+        } else if (counter === 2) {
+          // try modifying image1
+          expect(message).to.contain(path.basename(image1));
+          expect(currentStep).to.be(0);
+        } else if (counter === 3) {
+          // try modifying image2
+          expect(message).to.contain(path.basename(image2));
+          expect(currentStep).to.be(1);
+        } else if (counter === 4) {
+          // try modifying image3 (== image1)
+          expect(message).to.contain(path.basename(image3));
+          expect(currentStep).to.be(2);
+        } else if (counter === 5) {
+          // "already modified" message for image3
+          expect(message).to.contain(path.basename(image1));
+          expect(currentStep).to.be(2);
+        } else if (counter === 6) {
+          // "done" message
+          expect(message).not.to.be.empty();
+          expect(currentStep).to.be(3);
+        }
+      }
+    })(images)
+      .then(modifiedFiles => {
+        expect(counter).to.be(6);
+        expect(modifiedFiles).to.have.length(3);
+      }));
+  });
+
+  it("will properly process images in chunks even with erroneous images", function () {
+    this.timeout(30 * 1000);
+
+    const tmpDir = tmp.dirSync({unsafeCleanup: true});
+    const outPath = tmpDir.name;
+    const database = new Database(`${outPath}/database.json`);
+    const images = [fixtureFile, wrongImageFile, fixtureFile2];
+    const [image1, image2, image3] = images;
+
+    let onErrorCalled = false;
+    let counter = 0;
+
+    return cleanUpWhenDone(tmpDir)(modifyImages({
+      database,
+      key: "test",
+      outPath,
+      imageWidth: 10,
+      minify: true,
+      chunkSize: 2,
+      onError: (error, path) => {
+        onErrorCalled = true;
```

Comment:
Maybe test that the `onError` function is only called once. Will it be called for each erroneous file? Could be another test.